### ISSUE TITLE
Add text styles to multiple components

### ISSUE THREAD

**Issue by jeffmerten** [cerner/terra-clinical#262]
# Issue Description
There is a desire to apply the "attention" style to other UI elements, like in the example for allergies that are severe.
![attention allergy detail](https://user-images.githubusercontent.com/34603098/35590659-2716408e-05cd-11e8-9d40-e6b14742b488.png)

Currently ItemDisplay allows some text styles to be applied, but it's also desired to be in the DetailView title and also the LabelValueView value. 

## Issue Type
- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Consumers can apply the desired styling to the UI elements listed above in a manner similar to ItemDisplay.

**Comment by avinashg1994** [comment]
@neilpfeiffer, @bjankord 
I have the following queries regarding this enhancement:

1. `testStyle` for "itemDisplay" can take `'primary', 'secondary', 'strong', 'attention', or 'strikeThrough'` as a value.
So, Do we need to provide all these style options? 'or' Just the `attention` style?
2. In **clinical-detail-view**, the description specifies `textStyle` to be added for detailView title. So, do we have to add `textStyle` for "title, secondaryTitle and subTitles" or just "title". Also, do we have to add the same for "DetailView.DetailList" "title" as well?
3. For **label-value-view**, Should `textStyle` be added to the "Label" or "TextValue"?

**Comment by neilpfeiffer** [comment]
2 of the 3 items in the screenshot can already be achieved with current existing API (providing samples below), and item no. 3 is a duplicate request.

\
**1.  cells in the `terra-clinical-item-collection` summary view:**
The Clinical-Item-Collection already imports the Clinical-Item-Display as it's displays, so as a consumer, you just need to add the textStyle prop in the implementation. Sample:
```
<ItemCollection.Item
      comment=(<ItemCollection.Comment text="This is a comment on a p..." />)
>
      <ItemCollection.Display text="metFORMIN" textStyle="attention" />
      <ItemCollection.Display text="Severe" textStyle="attention" />
      <ItemCollection.Display text="Swelling" textStyle="secondary" />
      <ItemCollection.Display text="Drug" textStyle="secondary" />
</ItemCollection.Item>
```
(or) in `terra-table` --> provide ItemDisplay as table-cell content
(or) in `terra-list` --> provide ItemDisplay as list-item content

\
\
**2. Label-Value pairs in the `terra-clinical-detail-view` side panel:**
The Clinical-Detail-View already takes Clinical-Label-Value-Views as child component, so as a consumer, you just need to provide it a node as children instead of the text prop. Sample:
```
 <DetailView
    title="metFORMIN"
    details={[
       (<DetailView.DetailList key="Indicators">
             <DetailView.DetailListItem item=(<LabelValueView label="Severity"><ItemDisplay text="Severe" textStyle="attention" /></LabelValueView>) />
             <DetailView.DetailListItem item={detail1item2} />
             <DetailView.DetailListItem item={detail1item3} />
             <DetailView.DetailListItem item={detail1item4} />
         </DetailView.DetailList>),
     ]}
/>
```

\
\
**3. Header/Title as node element in the `terra-clinical-detail-view` side panel:**
Duplicated by #449 - adding a node instead of string for the title prop. Will be updated so that a consumer can provide a Clinical-Item-Display or the new Clinical-Results-Display component as the title. This part of the request will fully be resolved when the enhancement for #449 is made.

**Comment by neilpfeiffer** [comment]
Closing issue since only outstanding item is already encompassed in #449 and #492 as duplicate request.
